### PR TITLE
Talk while muted sound notification

### PR DESCRIPTION
**Pull request by damencho**

### CODE REVIEW

File: react/features/talk-while-muted/constants.js [hunk 1]
```diff
@@ -0,0 +1,14 @@
...
+ * @type {string}
+ */
+export const TALK_WHILE_MUTED_SOUND_ID = 'TALK_WHILE_MUTED_SOUND_ID';
+
+/**
+ * The file used for the talk while muted sound notification.
+ *
+ * @type {string}
+ */
+export const TALK_WHILE_MUTED_SOUND_FILE = 'talkWhileMuted.mp3';
```
Comment (hristoterezov) [inline react/features/talk-while-muted/constants.js]: Aren't we putting these kind of constants in sounds.js?

File: react/features/talk-while-muted/middleware.js [hunk 2]
```diff
@@ -0,0 +1,63 @@
+// @flow
+
+import { setCurrentNotificationUid } from './actions';
+import {
+    TALK_WHILE_MUTED_SOUND_FILE,
+    TALK_WHILE_MUTED_SOUND_ID
+} from './constants';
```
Comment (hristoterezov) [inline react/features/talk-while-muted/middleware.js]: I think those imports should be placed last, isn't it?

File: react/features/talk-while-muted/middleware.js [hunk 3]
```diff
@@ -0,0 +1,63 @@
...
+                const { currentNotificationUid } = getState()['features/talk-while-muted'];
+
+                if (currentNotificationUid && track.isAudioTrack() && track.isLocal() && !track.isMuted()) {
+                    dispatch(hideNotification(currentNotificationUid));
+                    dispatch(setCurrentNotificationUid());
+                }
+            });
+        conference.on(
+            JitsiConferenceEvents.TALK_WHILE_MUTED, () => {
+                const notification = dispatch(showNotification({
```
Comment (hristoterezov) [inline react/features/talk-while-muted/middleware.js]: Isn't it going to be safer to get the result directly from `showNotification` and than dispatch it?

I'm worried about async behaviour in the middlewares. Not sure what would be the result from the `dispatch` if we add async behaviour in a middleware in future.

WDYT?

File: react/features/talk-while-muted/constants.js (hunk 1 above)
Comment (damencho) [reply react/features/talk-while-muted/constants.js]: We have one in feature/chat and some in features/participants.

File: react/features/talk-while-muted/middleware.js (hunk 3 above)
Comment (damencho) [reply react/features/talk-while-muted/middleware.js]: Ok, I will change it. But, I just moved the code.

**Pull request merged**